Utils/FaviconFetcher.swift

```diff
@@ -13,108 +16,118 @@ class FaviconFetcherErrorType: ErrorType {
 /* A helper class to find the favicon associated with a url. This will load the page and parse any icons it finds out of it
  * If that fails, it will attempt to find a favicon.ico in the root host domain
  */
-class FaviconFetcher : NSObject, NSXMLParserDelegate {
+public class FaviconFetcher : NSObject, NSXMLParserDelegate {
+    public static var userAgent: String = ""
     static let ExpirationTime = NSTimeInterval(60*60*24*7) // Only check for icons once a week
 
-    private var siteUrl: NSURL // The url we're looking for favicons for
-    private var _favicons = [Favicon]() // An internal cache of favicons found for this url
-
     class func getForUrl(url: NSURL, profile: Profile) -> Deferred<Result<[Favicon]>> {
-        let f = FaviconFetcher(url: url)
-        return f.loadFavicons(profile)
-    }
-
-    private init(url: NSURL) {
-        siteUrl = url
+        let f = FaviconFetcher()
```

Comment:
Yeah, the way I originally wrote this the fetcher kept some state internally, so we needed separate instances. It could almost be class functions after this, but it does keep a little bit of state still. I'll see if I can remove it.